int32 width_l
int32 width_r
int32 height_t
int32 height_b